src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -13,28 +26,34 @@ public class Drivetrain extends SubsystemBase {
   private CANSparkMax rightMotor1;
   private CANSparkMax rightMotor2;
 
-  // TODO 2.1.1: Create DifferentialDrivetrainSim object (don't define it here)
+  DifferentialDrivetrainSim driveSim;
 
-  // TODO 2.2.1: Create gyro (AHRS)
+  AHRS gyro = new AHRS(SPI.Port.kMXP);
 
-  // TODO 2.2.3: Create DifferentialDriveKinematics
+  DifferentialDriveKinematics driveKinematics = new DifferentialDriveKinematics(DriveConstants.TRACK_WIDTH);
 
-  // TODO 2.2.4: Create DifferentialDrivePoseEstimator
+  DifferentialDrivePoseEstimator drivePose;
 
   // TODO 6.1.5: Create Feedforward and PIDs
 
 
   public Drivetrain() {
 
-    // TODO 1.1.2: Initialize motors
+    leftMotor1 = new CANSparkMax(DriveConstants.LEFT_MOTOR_1_ID,MotorType.kBrushless);
+    rightMotor1 = new CANSparkMax(DriveConstants.RIGHT_MOTOR_1_ID,MotorType.kBrushless);
 
-    // TODO 1.1.3: Set motors to brake mode
+    leftMotor1.setIdleMode(IdleMode.kBrake);
+    rightMotor1.setIdleMode(IdleMode.kBrake);
   
     // TODO 1.1.4: Make motor2s follow motor1s
 
-    // TODO 1.2.4: Invert motors if necessary
+    rightMotor1.setInverted(true);
+
+    if (!RobotBase.isReal()) {
+      driveSim = new DifferentialDrivetrainSim(DriveConstants.DRIVETRAIN_PLANT, DriveConstants.MOTOR, DriveConstants.GEAR_RATIO, DriveConstants.TRACK_WIDTH, DriveConstants.WHEEL_DIAMETER / 2, DriveConstants.MEASUREMENT_STD_DEVS);
+      drivePose = new DifferentialDrivePoseEstimator(driveKinematics, getGyroAngle(), getLeftPosition(), getRightPosition(), new Pose2d());
 
```

Comment:
drivePose should be defined outside the if statement so it will exist for both real and simulated robots.